diff --git a/autohotkey.php b/autohotkey.php
index f76e3dd..34415e5 100644
--- a/autohotkey.php
+++ b/autohotkey.php
@@ -289,9 +289,9 @@ $language_data = array (
     'STYLES' => array(
         'KEYWORDS' => array(
             1 => 'color: #AAAAFF; font-weight: bold;',       // reserved #blue
-            2 => 'color: #88FF88;',                         // BIV yellow
-            3 => 'color: #FF00FF; font-style: italic;',       // commands purple
-            4 => 'color: #888844; font-weight: bold;',       // functions #0080FF
+            2 => 'color: #444444; font-style: italic;',       // builtin variables #gray
+            3 => 'color: #1530AA; font-weight: bold;',       // commands purple
+            4 => 'color: #0088FF; font-weight: bold;',       // functions #0080FF
             5 => 'color: #000000; font-style: italic;',    // directives #black
             6 => 'color: #FF0000; font-style: italic;',      // hotkeys #red
             7 => 'color: #000000; font-style: italic;',    // gui commands #black
@@ -305,10 +305,10 @@ $language_data = array (
             0 => ''
             ),
         'BRACKETS' => array(
-            0 => 'color: #00FF00; font-weight: bold;'
+            0 => 'color: #00AA00; font-weight: bold;'
             ),
         'STRINGS' => array(
-            0 => 'font-weight: bold; color: #008080;'
+            0 => 'color: #FF0000;'
             ),
         'NUMBERS' => array(
             0 => 'color: #0000dd;'
@@ -344,7 +344,7 @@ $language_data = array (
         ),
     'URLS' => array(
         1 => '',
-        2 => 'http://www.autohotkey.com/docs/Variables.htm#{FNAME}',
+        2 => '',
         3 => 'http://www.autohotkey.com/docs/commands/{FNAME}.htm',
         4 => 'http://www.autohotkey.com/docs/Functions.htm#BuiltIn',
         5 => 'http://www.autohotkey.com/docs/commands/_{FNAME}.htm',
